### ISSUE TITLE
Loops in pytest

### ISSUE THREAD

**Issue by Yuri211200** [giecaruff/stoneforge#24]
Searching how to do loops with putest library in videos/sites etc.

**Comment by Yuri211200** [comment]
Opinion required for loops in porosity test. Any suggestions of changes in the code or ideas? Its in the ([tests_porosity](https://github.com/giecaruff/stoneforge/tree/porosity_test/stoneforge/tests_porosity)) branch on stoneforge @fabiojdf @fvizeus @rmotadutra @mmramos

**Comment by rmotadutra** [comment]
I think the best idea is the mix of @fabiojdf  and @mmramos ideas. The `np.random.uniform` method best represents the value that you get for porosity. For sure, if you have more elements it covers more possibilities. So, for me, the best idea is to use np.random.uniform with a size. For example: `np.random.uniform(low=0.0, high=1.0, size=250). I set 250 for the size but can be any value.

I don't like to use `np.linspace` because you'll get an array with evenly spaced numbers over the interval [0,1], i.e., the values of the array will be always increasing and this does not represent an array of porosities.

**Comment by mmramos** [comment]
@rmotadutra  I agree. We did some testing today with `np.linspace` + `np.meshgrid.` This procedure is very good for checking the possibility space, but it has a serious scaling problem because if it is a linspace of range 10, it will be 10 tests. If it's 2 linspaces it's 100 tests, for 3 it's 1000, for 5 it's 10 000, etc. And considering that each method has 10 equations, the code can easily reach 100 000 tests or more. And I'm really afraid that github might kick us out for this kind of practice. And the random test proves to be a viable (and recurrent) alternative.

about the equally distributed space, for MP oilfield for example, the smallest viable unit for the GR profile would have to be 0.0001, what would be a dx? Or dGR? Therefore, if we create a range of values ​​that varies from 0 to 150 with 0.00001 spacing, we would have an array with 1 500 000 elements that would represent all possible values. It is possible, well, it is. But will it create enmity with the server owners? Oh Yes. Soooooooooo I agree too.
